### PR TITLE
Re-enable dbt tests that connect to Postgres on Windows runner.

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

During the work on #2230  it was decided to disable the Windows dbt tests that require a Postgres connection in order to get that important fix through. This PR makes a Postgres database available on the Windows runner so that we can re-enable those tests 😄 

Simplified version of this article: https://www.cybertec-postgresql.com/en/postgresql-github-actions-continuous-integration/

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -185,6 +185,14 @@ jobs:
...
     name: DBT Plugin Python 3.8 Windows tests
     steps:
+    - name: Start PostgreSQL on Windows
+      run: |
+        $pgService = Get-Service -Name postgresql*
+        Set-Service -InputObject $pgService -Status running -StartupType automatic
+        Start-Process -FilePath "$env:PGBIN\pg_isready" -Wait -PassThru
+    - name: Create postgres user on Windows
+      run: |
+        & $env:PGBIN\psql --command="ALTER USER postgres PASSWORD 'password';" --command="\du"
```
Comment (jpy-git) [inline .github/workflows/ci-tests.yml:195]: sets up equivalent to:
```yml
dev:
      type: postgres
      host: localhost
      user: postgres
      pass: password
      port: 5432
      dbname: postgres
```

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -203,7 +211,7 @@ jobs:
       # Do not set explicitly temp dir for dbt as causes problems
       # None of these test need temp dir set
       run: |
-        python -m tox -e dbt018-winpy -- plugins/sqlfluff-templater-dbt -m "not dbt_connection_failure and not needs_dbt_connection"
+        python -m tox -e dbt018-winpy -- plugins/sqlfluff-templater-dbt
```
Comment (jpy-git) [inline .github/workflows/ci-tests.yml:214]: we don't need to exclude these anymore (removed these unused marks in the test code as well)

File: .github/workflows/ci-tests.yml [hunk 3]
```diff
@@ -185,6 +185,14 @@ jobs:
     runs-on: windows-latest
     name: DBT Plugin Python 3.8 Windows tests
     steps:
+    - name: Start PostgreSQL on Windows
+      run: |
+        $pgService = Get-Service -Name postgresql*
+        Set-Service -InputObject $pgService -Status running -StartupType automatic
+        Start-Process -FilePath "$env:PGBIN\pg_isready" -Wait -PassThru
+    - name: Create postgres user on Windows
```
Comment (jpy-git) [inline .github/workflows/ci-tests.yml:193]:
```suggestion
    - name: Set postgres user password
```

**Review by barrywhart — APPROVED** [review]

**Review by tunetheweb — APPROVED** [review]
Very good. Nice work!

**Pull request merged**